<h1 style='color:#5E7AFF'>CALENDARS</h1>

<strong>Registered on [JuliaHub](https://juliahub.com/ui/Packages/Calendars/yDHMq), 
    read the [Docs](https://docs.juliahub.com/Calendars/yDHMq), 
    get the source from the [repository](https://github.com/PeterLuschny/Calendars.jl) on GitHub, 
    or install with Pkg.add("Calendars").</strong>

In [1]:
using Calendars

┌ Info: Precompiling Calendars [2e69b1bb-2c9c-41e6-ab4b-04f4c0f206b8]
└ @ Base loading.jl:1478


<p style="color:brown;font-size:large">Conversion of Dates</p>

In [2]:
ConvertDate(("Gregorian", 1756, 1, 27), "Hebrew")

("AM", 5516, 11, 25)

In [3]:
ConvertDate("CE", 1756, 1, 27, "AM", true)

CE-1756-01-27 -> AM-5516-11-25


("AM", 5516, 11, 25)

<p style="color:brown;font-size:large">Date from day number</p>

In [4]:
DateFromDayNumber("Gregorian", 641027)

("CE", 1756, 1, 27)

<p style="color:brown;font-size:large">Day number from Date</p>

In [5]:
DayNumberFromDate("CE", 1756, 1, 27, true)

CE-1756-01-27 -> DN#0641027


641027

<p style="color:brown;font-size:large">Calendar Dates</p>

In [6]:
CalendarDates("Gregorian", 1756, 1, 27)

(("EC", 1756, 1, 27), ("CE", 1756, 1, 27), ("JD", 1756, 1, 16), ("AM", 5516, 11, 25), ("AH", 1169, 4, 24), ("ID", 1756, 5, 2), ("EN", 0, 0, 641029))

In [7]:
CalendarDates("CE", 1756, 1, 27, true)

European  EC-1756-01-27
Common    CE-1756-01-27
Julian    JD-1756-01-16
Hebrew    AM-5516-11-25
Islamic   AH-1169-04-24
IsoDate   ID-1756-05-02
EuroNum   EN#0641029


(("EC", 1756, 1, 27), ("CE", 1756, 1, 27), ("JD", 1756, 1, 16), ("AM", 5516, 11, 25), ("AH", 1169, 4, 24), ("ID", 1756, 5, 2), ("EN", 0, 0, 641029))

<p style="color:brown;font-size:large">Epoch of Diocletian, Coptic calendar</p>

In [8]:
CalendarDates("JD", 284, 8, 29, true)

European  EC-0284-08-29
Common    CE-0284-08-29
Julian    JD-0284-08-29
Hebrew    AM-4044-06-29
INVALID   00-0000-00-00
IsoDate   ID-0284-35-05
EuroNum   EN#0103607


(("EC", 284, 8, 29), ("CE", 284, 8, 29), ("JD", 284, 8, 29), ("AM", 4044, 6, 29), ("00", 0, 0, 0), ("ID", 284, 35, 5), ("EN", 0, 0, 103607))

<p style="color:brown;font-size:large">Epoch of Gregorian calendar (CE)</p>

In [9]:
CalendarDates("CE", 1582, 10, 15, true)

European  EC-1582-10-15
Common    CE-1582-10-15
Julian    JD-1582-10-05
Hebrew    AM-5343-07-19
Islamic   AH-0990-09-17
IsoDate   ID-1582-41-05
EuroNum   EN#0577738


(("EC", 1582, 10, 15), ("CE", 1582, 10, 15), ("JD", 1582, 10, 5), ("AM", 5343, 7, 19), ("AH", 990, 9, 17), ("ID", 1582, 41, 5), ("EN", 0, 0, 577738))

<p style="color:brown;font-size:large">Epoch of Islamic calendar (AH)</p>

In [10]:
CalendarDates("JD", 622, 7, 16, true)

European  EC-0622-07-16
Common    CE-0622-07-19
Julian    JD-0622-07-16
Hebrew    AM-4382-05-03
Islamic   AH-0001-01-01
IsoDate   ID-0622-29-05
EuroNum   EN#0227017


(("EC", 622, 7, 16), ("CE", 622, 7, 19), ("JD", 622, 7, 16), ("AM", 4382, 5, 3), ("AH", 1, 1, 1), ("ID", 622, 29, 5), ("EN", 0, 0, 227017))

<p style="color:brown;font-size:large">French Republican calendar</p>

In [11]:
CalendarDates("CE", 1792, 9, 22, true)

European  EC-1792-09-22
Common    CE-1792-09-22
Julian    JD-1792-09-11
Hebrew    AM-5553-07-06
Islamic   AH-1207-02-05
IsoDate   ID-1792-38-06
EuroNum   EN#0654417


(("EC", 1792, 9, 22), ("CE", 1792, 9, 22), ("JD", 1792, 9, 11), ("AM", 5553, 7, 6), ("AH", 1207, 2, 5), ("ID", 1792, 38, 6), ("EN", 0, 0, 654417))

<p style="color:brown;font-size:large">First crewed moon landing (Apollo 11)</p> 

In [12]:
CalendarDates("CE", 1969, 7, 20, true)

European  EC-1969-07-20
Common    CE-1969-07-20
Julian    JD-1969-07-07
Hebrew    AM-5729-05-05
Islamic   AH-1389-05-05
IsoDate   ID-1969-29-07
EuroNum   EN#0719000


(("EC", 1969, 7, 20), ("CE", 1969, 7, 20), ("JD", 1969, 7, 7), ("AM", 5729, 5, 5), ("AH", 1389, 5, 5), ("ID", 1969, 29, 7), ("EN", 0, 0, 719000))

<p style="color:brown;font-size:large">See the comment in README.md for the next example, that may seem confusing at first glance.</p> 

In [13]:
PrintEuropeanMonth(1582, 10)

# Calendars of the month EC-1582-10
|    Common     |   European    |    Julian     |    Hebrew     |    Islamic    |    IsoDate    |  EuroNum   | JulianNum  |
| :---:         |  :---:        | :---:         |  :---:        | :---:         |  :---:        | :---:      | :---:      |
| CE-1582-10-11 | EC-1582-10-01 | JD-1582-10-01 | AM-5343-07-15 | AH-0990-09-13 | ID-1582-41-01 | EN#0577734 | JN#2299156 |
| CE-1582-10-12 | EC-1582-10-02 | JD-1582-10-02 | AM-5343-07-16 | AH-0990-09-14 | ID-1582-41-02 | EN#0577735 | JN#2299157 |
| CE-1582-10-13 | EC-1582-10-03 | JD-1582-10-03 | AM-5343-07-17 | AH-0990-09-15 | ID-1582-41-03 | EN#0577736 | JN#2299158 |
| CE-1582-10-14 | EC-1582-10-04 | JD-1582-10-04 | AM-5343-07-18 | AH-0990-09-16 | ID-1582-41-04 | EN#0577737 | JN#2299159 |
| CE-1582-10-15 | EC-1582-10-15 | JD-1582-10-05 | AM-5343-07-19 | AH-0990-09-17 | ID-1582-41-05 | EN#0577738 | JN#2299160 |
| CE-1582-10-16 | EC-1582-10-16 | JD-1582-10-06 | AM-5343-07-20 | AH-0990-09-18 | ID-1582-41-06 

# Calendars of the month EC-1582-10
|    Common     |   European    |    Julian     |    Hebrew     |    Islamic    |    IsoDate    |  EuroNum   | JulianNum  |
| :---:         |  :---:        | :---:         |  :---:        | :---:         |  :---:        | :---:      | :---:      |
| CE-1582-10-11 | EC-1582-10-01 | JD-1582-10-01 | AM-5343-07-15 | AH-0990-09-13 | ID-1582-41-01 | EN#0577734 | JN#2299156 |
| CE-1582-10-12 | EC-1582-10-02 | JD-1582-10-02 | AM-5343-07-16 | AH-0990-09-14 | ID-1582-41-02 | EN#0577735 | JN#2299157 |
| CE-1582-10-13 | EC-1582-10-03 | JD-1582-10-03 | AM-5343-07-17 | AH-0990-09-15 | ID-1582-41-03 | EN#0577736 | JN#2299158 |
| CE-1582-10-14 | EC-1582-10-04 | JD-1582-10-04 | AM-5343-07-18 | AH-0990-09-16 | ID-1582-41-04 | EN#0577737 | JN#2299159 |
| CE-1582-10-15 | EC-1582-10-15 | JD-1582-10-05 | AM-5343-07-19 | AH-0990-09-17 | ID-1582-41-05 | EN#0577738 | JN#2299160 |
| CE-1582-10-16 | EC-1582-10-16 | JD-1582-10-06 | AM-5343-07-20 | AH-0990-09-18 | ID-1582-41-06 | EN#0577739 | JN#2299161 |
| CE-1582-10-17 | EC-1582-10-17 | JD-1582-10-07 | AM-5343-07-21 | AH-0990-09-19 | ID-1582-41-07 | EN#0577740 | JN#2299162 |
| CE-1582-10-18 | EC-1582-10-18 | JD-1582-10-08 | AM-5343-07-22 | AH-0990-09-20 | ID-1582-42-01 | EN#0577741 | JN#2299163 |
| CE-1582-10-19 | EC-1582-10-19 | JD-1582-10-09 | AM-5343-07-23 | AH-0990-09-21 | ID-1582-42-02 | EN#0577742 | JN#2299164 |
| CE-1582-10-20 | EC-1582-10-20 | JD-1582-10-10 | AM-5343-07-24 | AH-0990-09-22 | ID-1582-42-03 | EN#0577743 | JN#2299165 |
| CE-1582-10-21 | EC-1582-10-21 | JD-1582-10-11 | AM-5343-07-25 | AH-0990-09-23 | ID-1582-42-04 | EN#0577744 | JN#2299166 |
| CE-1582-10-22 | EC-1582-10-22 | JD-1582-10-12 | AM-5343-07-26 | AH-0990-09-24 | ID-1582-42-05 | EN#0577745 | JN#2299167 |
| CE-1582-10-23 | EC-1582-10-23 | JD-1582-10-13 | AM-5343-07-27 | AH-0990-09-25 | ID-1582-42-06 | EN#0577746 | JN#2299168 |
| CE-1582-10-24 | EC-1582-10-24 | JD-1582-10-14 | AM-5343-07-28 | AH-0990-09-26 | ID-1582-42-07 | EN#0577747 | JN#2299169 |
| CE-1582-10-25 | EC-1582-10-25 | JD-1582-10-15 | AM-5343-07-29 | AH-0990-09-27 | ID-1582-43-01 | EN#0577748 | JN#2299170 |
| CE-1582-10-26 | EC-1582-10-26 | JD-1582-10-16 | AM-5343-07-30 | AH-0990-09-28 | ID-1582-43-02 | EN#0577749 | JN#2299171 |
| CE-1582-10-27 | EC-1582-10-27 | JD-1582-10-17 | AM-5343-08-01 | AH-0990-09-29 | ID-1582-43-03 | EN#0577750 | JN#2299172 |
| CE-1582-10-28 | EC-1582-10-28 | JD-1582-10-18 | AM-5343-08-02 | AH-0990-09-30 | ID-1582-43-04 | EN#0577751 | JN#2299173 |
| CE-1582-10-29 | EC-1582-10-29 | JD-1582-10-19 | AM-5343-08-03 | AH-0990-10-01 | ID-1582-43-05 | EN#0577752 | JN#2299174 |
| CE-1582-10-30 | EC-1582-10-30 | JD-1582-10-20 | AM-5343-08-04 | AH-0990-10-02 | ID-1582-43-06 | EN#0577753 | JN#2299175 |
| CE-1582-10-31 | EC-1582-10-31 | JD-1582-10-21 | AM-5343-08-05 | AH-0990-10-03 | ID-1582-43-07 | EN#0577754 | JN#2299176 |
| CE-1582-11-01 | EC-1582-11-01 | JD-1582-10-22 | AM-5343-08-06 | AH-0990-10-04 | ID-1582-44-01 | EN#0577755 | JN#2299177 |
| CE-1582-11-02 | EC-1582-11-02 | JD-1582-10-23 | AM-5343-08-07 | AH-0990-10-05 | ID-1582-44-02 | EN#0577756 | JN#2299178 |
| CE-1582-11-03 | EC-1582-11-03 | JD-1582-10-24 | AM-5343-08-08 | AH-0990-10-06 | ID-1582-44-03 | EN#0577757 | JN#2299179 |
| CE-1582-11-04 | EC-1582-11-04 | JD-1582-10-25 | AM-5343-08-09 | AH-0990-10-07 | ID-1582-44-04 | EN#0577758 | JN#2299180 |
| CE-1582-11-05 | EC-1582-11-05 | JD-1582-10-26 | AM-5343-08-10 | AH-0990-10-08 | ID-1582-44-05 | EN#0577759 | JN#2299181 |
| CE-1582-11-06 | EC-1582-11-06 | JD-1582-10-27 | AM-5343-08-11 | AH-0990-10-09 | ID-1582-44-06 | EN#0577760 | JN#2299182 |
| CE-1582-11-07 | EC-1582-11-07 | JD-1582-10-28 | AM-5343-08-12 | AH-0990-10-10 | ID-1582-44-07 | EN#0577761 | JN#2299183 |
| CE-1582-11-08 | EC-1582-11-08 | JD-1582-10-29 | AM-5343-08-13 | AH-0990-10-11 | ID-1582-45-01 | EN#0577762 | JN#2299184 |
| CE-1582-11-09 | EC-1582-11-09 | JD-1582-10-30 | AM-5343-08-14 | AH-0990-10-12 | ID-1582-45-02 | EN#0577763 | JN#2299185 |
| CE-1582-11-10 | EC-1582-11-10 | JD-1582-10-31 | AM-5343-08-15 | AH-0990-10-13 | ID-1582-45-03 | EN#0577764 | JN#2299186 |